tests/kani/Panic/const.rs

```diff
@@ -0,0 +1,16 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! This test checks that `panic!` can be used in a const fn
+
+const fn my_const_fn() {
+    panic!()
+}
+
+#[kani::proof]
+pub fn check_something() {
+    let x: u8 = if kani::any() { 3 } else { 95 };
+    if x > 100 {
+        my_const_fn();
```

Comment:
This call isn't reachable (`x > 100` is always false). The test shows you can use `panic!` in a `const fn` but I think you want to also call it, right?

Reply:
The impetus for this was definitely a compile-time error, so maybe not

Reply:
Correct, I'm intentionally not calling it so that the test passes. As Ted pointed out, compilation currently fails. The purpose of the `kani::any()` is to make sure the compiler doesn't optimize the function out (so we still have to codegen it).